darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterManagerImpl.java

```diff
@@ -102,9 +128,10 @@ public boolean handleDarkRequest(RestRequest originalRequest, RequestContext ori
         for (String darkClusterName : configMap.keySet())
         {
           RestRequest newD2Request = rewriteRequest(reqCopy, darkClusterName);
+          RestRequest d2RequestWithHeaders = addDarkRequestHeaders(darkClusterName, newD2Request);
```

Comment:
can you try to add an argument to the rewriteRequest to add the header(s)? Otherwise we are producing intermediate garbage by creating the request and then throwing it away after copying it to create another request.